src/lib/redux/slices/pitwallSlice/models.ts

```diff
@@ -249,6 +283,55 @@ export interface TimingTelemetry {
   lapDistancePercentage: number;
 }
 
+export class TimingTelemetryClass {
+  currentLap!: number;
+  incidents!: number;
+  completeLaps!: number;
+  currentLapTime!: number;
+  deltaToSessionBest!: number;
+  lapDistancePercentage!: number;
+
+  constructor(timingTelemetry: TimingTelemetry | null | undefined) {
```

Comment:
What do you think about making the timing telemetry state type in redux this class? That way if other components need it, they don't have to worry about instantiation or anything. Also not sure about memory or performance overhead of creating so many class objects if it were used in many places. At least with the redux state approach it would only be instantiated once per update.

Reply:
I have now merged the interface and class for the TimingTelemetry.